### PR TITLE
fix(build): make e2e tests faster

### PR DESCRIPTION
**Pull request by marclaval**
While trying to put more in SauceLabs, I found out the e2e tests take 20+ minutes in our builds, mainly because protractor waits a lot.
Trying something to improve that.

### CODE REVIEW

**Comment by mlaval** [comment]
It works!
The build time is reduced from ~32mins to ~15mins for Dart jobs and ~12mins for JS job.
The reason behind it is that each of the 92 specs ran with Protractor was waiting 14s, so that was a flat waiting time of 21+ mins.

The fix is quite hacky as it waits for the "Loading..." string to disappear from the DOM.
But I'm not familiar enough with ourbenchmarking process to validate it.
@tbosch @juliemr @yjbanov Can you please review?

**Comment by mhevery** [comment]
@mlaval nice work sir!

**Comment by tbosch** [comment]
@mlaval We should have a properly working API to wait for Angular now. Could you work on using this instead (see https://github.com/angular/angular/issues/3829)?

**Comment by mhevery** [comment]
Assigned to @mlaval as a follow up.

File: protractor-shared.js [hunk 1]
```diff
@@ -192,7 +192,7 @@ function patchProtractorWait(browser) {
   var sleepInterval = process.env.TRAVIS || process.env.JENKINS_URL ? 14000 : 8000;
   browser.get = function() {
     var result = _get.apply(this, arguments);
-    browser.sleep(sleepInterval);
+    browser.driver.wait(protractor.until.elementLocated(By.js('var cs = document.body.children; var isLoading = false; for (var i = 0; i < cs.length; i++) {if (cs[i].textContent.indexOf("Loading...") > -1) isLoading = true; } return !isLoading ? document.body.children : null')), sleepInterval);
```
Comment (gdi2290) [inline protractor-shared.js]: TIL you can also pass in a function rather than a string

**Pull request closed without merging**